src/concurrency/Sync.java

```diff
@@ -21,11 +21,33 @@ public void run() {
         }
     }
 
-    public static void main(String[] args)  {
+    public static void main(String[] args) throws InterruptedException {
         Sync syncThread = new Sync("sync thread");
         syncThread.start();
+        syncThread.join();
+        System.out.println("Thread 1: " + syncThread.nums);
+        System.out.println("Very cool that IntelliJ helps you insert the interrupt exception in the method");
+
+        Sync syncThread2 = new Sync("sync thread2");
+        syncThread2.start();
```

Comment:
The Bonus exercise included making the 'nums' field a shared resource.  In your solution, it is an instance field, meaning that each thread has it's own 'nums' field.  Adding the 'static' keyword to the 'nums' field results in each thread adding another set of random numbers to the list.  The code was to safeguard the shared resource, such that each thread would wait its turn to use the resource, then generate a completely new set of data (exclusive of 'cached' values).